Licensing Portal: implement the unassign licenses feature

wilmersondasilva opened the pull request:
#### Proposed Changes

* Licensing Portal: add the possibility to unassign a license from a site. Thus the users can unlinking a license from a specific site and reuse it on another site.

#### Testing Instructions

* Make sure you have a wordpress.com website.
* Open up Licensing -> Licenses.
* Issue a new license, if don't have one.
* Open some license details. If this license is not assigned to some site, please do it.
* The "Unassign License" button should appear on the bottom left corner.
* Click on the "Unassign License" button. An alert modal should appear with information about this action and ask for you to confirm that want to unassign that license from the site.

#### Screenshots
![image](https://user-images.githubusercontent.com/5550190/182459485-440f2cba-f413-4b78-805b-578d499e2e20.png)

![image](https://user-images.githubusercontent.com/5550190/182459596-5a93412e-a2c0-48aa-b1e3-ae2b241f43dd.png)


#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React, or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have you sent any new strings for translation (PCYsg-1vr-p2) ASAP?

Related to #

wilmersondasilva committed:
Licensing Portal: create use-unassign-license-mutation hook

This hook will make the HTTP request that unassign one license from a site.

wilmersondasilva committed:
Licensing Portal: create use-unassign-license-mutation hook

This hook will make the HTTP request that unassign one license from a site.

wilmersondasilva committed:
Licensing Portal: create use-unassign-license-dialog component to users confirm they want to unassign a license

wilmersondasilva committed:
Licensing Portal: create use-unassign-license-dialog component to users confirm they want to unassign a license

wilmersondasilva committed:
Licensing Portal: add a button to unassign licenses

Currently, it is only possible to revoke licenses, and revoked licenses can't be used again. To allow the process of unlinking a license from a specific site, an "Unassign License" button was added.

wilmersondasilva committed:
Licensing Portal: add a button to unassign licenses

Currently, it is only possible to revoke licenses, and revoked licenses can't be used again. To allow the process of unlinking a license from a specific site, an "Unassign License" button was added.

bangank36 committed:
Unified design picker: Event tracking (#66005)

- Add event when user exit preview
- Update category select event
- Remove scroll event on mobile for UDP

jimmy319 committed:
Import: redirecting to the themes page for non-WP transfers (#65926)

* Enable feature redirect-to-themes on dev

* Make doneButton show different labels based on the feature flag

* Feature flag based redirection for import content only screen

* Feature flag based redirection for blogger importer complete screen

* Feature flag based redirection for Medium importer complete screen

* Feature flag based redirection for Squarespace importer complete screen

* Feature flag based redirection for Wix importer complete screen

* Set the label of DoneButton of import everything screen explicitly

bangank36 committed:
Unify loading screens in all onboarding flows (#65456)

- Persist the intent data by storing in STEPPER_INTERNAL_STORE
- Fix loading messages for sell flow
- Fix loading messages for difm flow
- Fix loading screen for import flow

mreishus committed:
step-container: Stop blocking click events (Redux) (#66019)

andrii-lysenko committed:
update: dont use refetch when unnecessary. (#65890)

mreishus committed:
Add Premium badge to premium themes in the design preview header (#65944)

sirbrillig committed:
Checkout: Change term variant picker discount display to compare against selected price (#65998)

* Add unformatted price and term to useGetProductVariants result

* Calculate discount inside ItemVariantPrice

* Make selectedVariant optional in ItemVariantPrice

* Pass selectedVariant to ItemVariantPrice

* Fix discountPercentage calculation

* Simplify ItemVariantOption props

* Rename prices inside WPCOMProductVariant to be more clear

* Set selectedVariant based on selected item rather than highlighted

* Simplify calculation of selectedVariantIndex

* Adjust variant picker tests to cover new behavior

* Renamme selectedVariant to compareTo in ItemVariantPrice

* Change compareTo variable name to be more flexible

andres-blanco committed:
set flag to true (#66052)

worldomonation committed:
Specs: (#66000)

- remove dead specs for Gutenboarding - no traffic is currently going there.

wilmersondasilva committed:
Licensing Portal: fix the Assign License button style

In some resolutions, the Assign License was not being shown.

wilmersondasilva committed:
Licensing Portal: fix the Assign License button style

In some resolutions, the Assign License was not being shown.

vykes-mac committed:
Sites Dashboard: Site thumbnail component (#65868)

* Create site thumbnail component.

paulopmt1 committed:
Improved string translations strategy on PR template (#66062)

sirbrillig committed:
Checkout: use post-checkout pending page for non-redirect payment methods (#65273)

* Change useCreatePaymentCompleteCallback to redirect to pending page

* WIP: add displayRenewalSuccessNotice to CheckoutPending

* Add receipt data from D84365-code to receipt object

* Remove period, expiry, and email from renew success notice

* Remove renew success notice from useCreatePaymentCompleteCallback

* Fetch receipt data on pending page

* Add non-renewal success notice to pending page

* Remove post-checkout success message on customer home page

* Remove post-checkout success message on theme-upload page

* Make all pending page notices displayOnNextPage

* Move pending page notices to function

sirbrillig committed:
Checkout: Remove pending URL from getThankYouPageUrl (#65417)

* Remove unused transactionResult from useGetThankYouUrl

* Remove PendingOrderPathFragment from getThankYouPageUrl

* Update getThankYouPageUrl to allow for no receiptId at all

* Update getThankYouPageUrl tests for receipt placeholder and no pending

* Rename getPendingOrReceiptId to be more accurate

* Update business upsell test to have it expect upsell with placeholder

worldomonation committed:
TeamCity: (#66058)

- reduce frequency of AuthenticationE2E to every 3 hours.

wojtekn committed:
Sort Sites Dashboard by Last Publish (#65882)

Co-authored-by: Philip Jackson <p-jackson@live.com>

p-jackson committed:
Sites Dashboard: Rename <TabPanel> to <SitesTableTabPanel> (#65876)

arcangelini committed:
Signup: add newsletters flow (#66082)

* Signup: add new flow for Newsletters

* Update redirect to stepper

* Remove signup-header__left style

* Forgot to add siteSlug in query param

arcangelini committed:
Fix applied (#66088)

renancarvalho committed:
Help Center: Add site type check on help center (#66018)

* Add site type check on help center

* Fix typescript compile error

* Refactor isSimpleSite redux selector

renancarvalho committed:
Add basic link in bio intro step (#66086)

* Add basic link in bio intro step

* Use SVG BG instead

* Add translations

Co-authored-by: Omar Alshaker <omar@omaralshaker.com>

leonardost committed:
Domains: Fix notice for expired domains that are in active auction (#66067)

* Add aftermarket auction start and end dates to domain object

* Fix unit tests

sirbrillig committed:
Plans: Change plan price selectors to remove lodash get (#66069)

* Change plan price selectors to remove lodash get

* Add null to fix return type of selectors

* TS Fixes for corrected return type of selectors

* Handle null plan prices differently in usePlanPrices

sirbrillig committed:
Plans: Update "free domain with plan" sidebar upsell to go to domains first (#66022)

* Redirect from domain search page to plans page when in upsell flow

* Redirect to checkout from plans page with domain if in upsell flow

* Rename addDomainFlow query param to domainAndPlanPackage

* Rename withProduct to domainAndPlanPackage

* Hide domain search plan upsell during upsell

If we are already in the upsell flow, there's no need to show another
upsell banner.

* Add a banner to the plans page when in the domain and plan flow

* Refactor to add the domain when it is clicked

* Pass domainAndPlanUpsellFlow to getDomainPriceRule

* Update banner to re-use translation strings

Co-authored-by: Michael Cain <michael@michaeldcain.com>

creativecoder committed:
Fix `fetchPriority` prop to prevent React warning (#66070)

* Prevent React warning about `fetchPriority` prop
* Update variable name, as well, for consistency

lupus2k committed:
Marketplace: Move components into it's own modules (#66060)

* Refactor plugins to components

alshakero committed:
Help Center: Fix events bugs and increase the size of the user pool to 30% (#66053)

benzhu56 committed:
My Home: Add Promote Post to my home page swipeable (#65996)

* Add promote post to my home swipeable (#65809)

* Use wpcom-proxy for requests

Co-authored-by: Ben Zhu <ben.zhu@a8c.com>

* rename dsp to promote-post for clarity

* Fix Codestyle failure:
"Don’t use literal URLs without angle brackets (no-literal-urls) (md/remark)"

Co-authored-by: Ben Zhu <6070516+benz1924@users.noreply.github.com>
Co-authored-by: Ben Zhu <ben.zhu@a8c.com>
Co-authored-by: James Gill <james.gill@automattic.com>

taipeicoder committed:
Support site_tagline in design previews (#66074)

taipeicoder committed:
Do not set site title if users have not provided one during onboarding (#66078)

gmovr committed:
Show domain USP in marketplace only if the billing interval is yearly (#66081)

* Added a condition to only show the 'free domain for one year' USP when a user is on Free AND has toggled the pricing interval to 'yearly'

* Updated the support text to not include hour/day coverage

claudiucelfilip committed:
Remove masterbar upsell button (#65728)

flootr committed:
`DomainSuggestion`: refactor tests to `@testing-library/react` (#65967)

bangank36 committed:
Update spacing for design preview title (#66056)

- Increase space for udp title
- Provide spacing update for flow header including logo, header, back button

escapemanuele committed:
Help Center: Show in all calypso (#65888)

alshakero committed:
Stepper: Make intro step generic and use it in Newsletters flow (#66102)

* Make intro step generic

* Capitalize step component name

* Use className instead of abusing stepName

arcangelini committed:
Signup: add link-in-bio flow (#66090)

* Working version

* Signup: added link-in-bio flow

mreishus committed:
ETK: Starter_Page_Templates: Only calculate cache key when needed (#66095)

olaseni committed:
Professional Email: Hide optional fields in the "Add mailbox" form for consistency with the Email Comparison view (#66106)

* Export password reset tip field as a standalone component

* Hide redundant fields in email form to reduce user journey

* Add default value for password reset field

* Import variables css

escapemanuele committed:
Fix for css targeting (#66129)

sirbrillig committed:
Checkout: Refactor post-checkout URL generation to clarify behavior (#65947)

* Move cookie URL changes to own function

* Refactor renewal url block to simplify logic

* Do not update cookies for manual renewals

* Refactor new blog URL handling to clarify effect

* Remove fallback url usage for new blog flow

* Unify new blog conditions

* Simplify traffic-guide url generation

* Add receipt_for_domain to ResponseCartProductExtra

* Determine if thank-you URL is eligible for signup cookie inline

* Remove unused isEligibleForSignupDestination

* Fix doesCartContainGoogleAppsWithoutDomainReceipt condition

* Fix doesCartContainGoogleAppsWithoutDomainReceipt

* Update tests to replace isEligibleForSignupDestination

* Move fallback URL generation into updateUrlInCookie

* Move fallbackUrl generation closer to usage

* Simplify noPurchaseMade flow

* Use early returns to simplify getFallbackDestination

* Prevent undefined from being injected into receipt URLs

* Remove unnecessary getSimpleThankYouUrl

* Remove unnecessary generic checkout page URL

* Allow default URL to include receipt ID placeholder in tests

* Export arguments to getThankYouPageUrl

* Remove orderId from calls to getThankYouPageUrl

heydemoura committed:
Add click event to Card and control FormRadio from state (#65980)

heydemoura committed:
Jetpack Licensing: Update license revoke dialog to make it fullscreen on mobile and some extra tweaks (#65988)

* Fix dialog button width on mobile

* Update mobile dialog to be fullscreen

* Upadte spacing between dialog action buttons

fullofcaffeine committed:
[GB 13.7.x] Open Site Editor in the Gutenframe by using the `site-editor.php` route (#65997)

* Change the Site Editor route to site-editor.php to match the changes in Gutenberg 13.7

* Update route in the `getSiteEditorUrl` helper

worldomonation committed:
chore: fix more ESLint errors and warnings. (#66096)

* ESlint fixes

* settings.kts
- define environment variables using param construct
- add TIMING environment variable

* packages/calypso-mp-data-analysis/package.json
- fix license

* eslintrc
- remove dead rule about inclusive language

* client/state/utils
- fix typo

agrullon95 committed:
Replace /block-editor with /themes?status=active endpoint (#66028)

* Replace /block-editor with /themes endpoint

* Remove isFSEActive usage in signup flows

* Remove is_fse_active from blockEditorSettings

* Remove isFSEActive from DIFM flow

* Return /theme-setup response inside conditiona;

worldomonation committed:
Plans Revert: upgrade plan from Free to Premium on existing site with content. (#65912)

* RestAPICLient:
- add method to create a post on site.
- add method to upload media to the site.

* PlansPage:
- narrow the button selectors down for selecting the plan.

Spec:
- add beforeAll hook to add test posts to the site.
- perform checks to see if the posts remain.

* Spec:
- replace hardcoded value of Premium in describe block title

* Spec:
- update file name

PlansPage:
- add comment on why the "first" locator is specified.

* RestAPIClient:
- implement local file upload

Yarn:
- add new dependency 'form-data'

* Spec:
- add steps to validate the uploaded image item.
- add more information logging.

MediaPage:
- implement branching logic for the selectItem method to support selection by name.

APIDeleteSite:
- add logging.

* RestAPIClient:
- remove unnecessary change in the header generation method

* Spec:
- remove commented out and testing code

* RestAPIClient:
- add tests for createPost

* Spec:
- use beforeAll hook instead

sirbrillig committed:
Checkout: Improve pending page UI (#66101)

* Move pending page content to own component

* Replace loading content with LoadingEllipsis

* Move pending page to own directory

* Move CheckoutPending styles to scss file

* Add wp-brand-font to title text

* Make error notices persistent

displayOnNextPage does not appear to actually work.

* Remove max-width on pending content wrapper

CodeyGuyDylan committed:
Add: Backup Retention Disclaimer Information (#65572)

* Add or remove hashes when FAQ's are opened or closed

We want to be able to link to FAQ's on the jetpack-connect pricing page and have them be open when linked to. Under the circumstance that somebody were to go to the link with a hash already in the URL, and then click a link that leads to the same FAQ, the FAQ would not be opened since no change in hash was detected. To prevent this, these changes remove the hash anytime the FAQ is closed, so if it were to be linked to again, it would then detect a hash changed and open up the FAQ. The hash's are also added to the URL when the FAQ is opened for user convenience if they were to want to send that link to somebody else with the FAQ already open

* Give component ability to automatically open if it was linked to via URL hash

* Add disclaimer to bottom of card

* Add disclaimer to product object

* Add asterisk to end of backup feature and move it to the bottom

* add asterisks to backup cards on upgrade page

* Dynamically generate link

If the FAQ is already on the page, we want to link to it as an anchor link, not in a separate page. So we check for the ID on the page, and if it exists, we just use an anchor link to jump to it on the page. Otherwise a full URL gets sent as the link and it opens a new window

* Set getter to get product disclaimer

* Add type for feature item

* configure disclaimer translation to display only if the corresponding feature is active

* Add disclaimer to plan cards

* Override disclaimer in upgrade product cards

* Remove 'here' from FAQ

candy02058912 committed:
Add blocks and editor tags to the Design category (#66151)

aneeshd16 committed:
DIFM Landing: Increment counter on page view (#66080)

gcsecsey committed:
enable the `marketplace-jetpack-plugin-search` flag on wpcalypso env (#66133)

flootr committed:
`CheckoutThankYouHeader`: refactor tests to `@testing-library/react` (#66108)

* `CheckoutThankYouHeader`: refactor tests to `@testing-library/react`

* also assert for length of messages

flootr committed:
`CheckoutThankYou`: refactor tests to `@testing-library/react` (#66114)

* `CheckoutThankYou`: refactor tests to `@testing-library/react`

* normalize cases for mocks

valterlorran committed:
adding price information behind the feature flag (#66135)

valterlorran committed:
Theme Showcase: Updating design for the theme popover on the themes page (#65766)

* Updating design for the theme popop on the themes page

* Messages should reflect the state of the site or theme(purchased or not)

* Redirect to checkout when clicking on the link inside the popover

* Adapting the theme component for testing

* Adding tests for premium theme popover

* Removing mention to Pro plan

* Missing border spacing when the theme was active

* Fixing spelling error

zaguiini committed:
Social Sign Up: Warn the user about an already existing account (#65908)

Co-authored-by: Daniel Bachhuber <daniel.bachhuber@automattic.com>
Co-authored-by: Philip Jackson <p-jackson@live.com>

wojtekn committed:
Fix conditions that prevent from editing or removing the site's owner (#65966)

* Fix conditions that prevent from editing or removing the site's owner

Currently, there are conditions that prevent from displaying UX
that allows editing or removing the site's owner. Those do not include
Jetpack sites, and this PR modifies those to include Jetpack sites too.

* Modify code that prevents from deleting owner to not catch self-hosted Jetpack

Admin user can delete other admin users on the self-hosted Jetpack sites.
I modify the logic that prevents from removing owner to catch only Atomic
and non-Jetpack sites.

flootr committed:
`PlanTypeSelector`: refactor tests to `@testing-library/react` (#66116)

* `PlanTypeSelector`: refactor tests to `@testing-library/react`

* use `.toBeChecked` and assert length of radios

cpapazoglou committed:
Marketplace: premium plugins now require a personal plan. (#66122)

* Marketplace all plans: adds feature flag.

* Marketplace on personal: removes pro plan - adds personal.

* Marketplace: removes collect payments feature from personal plan.

* Marketplace on personal: gets billing interval from state.

* Marketplace on personal: correctly add monthly plan.

* Checkout: respects 0 as a possible variant index.

Co-authored-by: cpap <papazoglou.charalampos@gmail.com>

sirbrillig committed:
Checkout: Rename CheckoutSystemDecider and CompositeCheckout (#66103)

* Rename CheckoutSystemDecider to CheckoutMainWrapper

* Rename CompositeCheckout to CheckoutMain

* Rename CheckoutMain tests file

zaguiini committed:
Social Login: Offer to create a new account if it doesn't exist (#65909)

Co-authored-by: Daniel Bachhuber <daniel.bachhuber@automattic.com>
Co-authored-by: Philip Jackson <p-jackson@live.com>

robertf4 committed:
Jetpack Agency Dashboard: Use jetpack_partner_types to detect agencies (#65041)

edanzer committed:
E2E Testing: Reenable e2e tests for adding template parts

wilmersondasilva committed:
Licensing Portal: update the Unassign License dialog design

The design/copy was the same as the Revoke License dialog. This commit changes some colors and updates the copy for one more suitable for the unassign action.

wilmersondasilva committed:
Licensing Portal: update the Unassign License dialog design

The design/copy was the same as the Revoke License dialog. This commit changes some colors and updates the copy for one more suitable for the unassign action.

rcanepa committed:
Jetpack Agency dashboard: enable plugin management in calypso.live (#66168)

* Enable plugin management in horizon env

* Enable plugin management section in horizon

worldomonation committed:
E2E: update `Signup: Paid` spec to cover the FTME and Launch flows. (#66064)

* RestAPIClient:
- deprecate the SiteDetails interface.

* StartSiteFlow
- define new type for Goals
- additional steps in `selectGoal` to select the dropdown item.

GeneralSettingsPage
- add `visit` method.

Signup POMs
- move into its own folder.
- rename where possible to a standard `SignupXYZ` schema.

SignupPickPlanPage
- change return signature of `selectPlan` method to NewSiteResponse.

* Signup: Free, Plans: Business
- change variable typing.

* FTME: Personal
- rework the Signup: Paid plan to FTME: Personal spec.
- covers the following: user signup, plan selection, checkout coupon, onboarding flow, write, launch flow.

* Spec:
- rename to FTME: Sell

MyHomePage:
- change selector for the top card message

StartSiteFlow:
- unify selector for `selectTheme`
- remove unused method

* Signup: Free
- skip a step

* StartSiteFlow:
- rename selector

* Spec:
- remove testing code

alshakero committed:
Restrict help-center CSS to the help center (#66181)

j6ll committed:
Add promote posts button to stats entries, with feature flag (#66024)

* Add promote post to my home swipeable (#65809)

* Add promote post to my home swipeable

* Use wpcom-proxy for requests

Co-authored-by: Ben Zhu <ben.zhu@a8c.com>

* rename dsp to promote-post for clarity

* Adds the promote button to the site stats post list

The button appears after the "external" link icon - these both appear on hovering the post title

* Update stats unit tests to reflect id addition

* Adds tracks event for the promote post button

Co-authored-by: Ben Zhu <6070516+benz1924@users.noreply.github.com>
Co-authored-by: Ben Zhu <ben.zhu@a8c.com>
Co-authored-by: James Gill <james.gill@automattic.com>

elliottprogrammer committed:
Jetpack post-connection: Redirect secondary user back to wp-admin. (#65861)

kthai15 committed:
Add/dsp widget entrypoint tracking (#66136)

* add tracking on widget start for a specified entry point

* add tracking to Promote Posts home page swipeable

enejb committed:
App Banner: Remove the banner experiemnt (#65860)

Apply the treatment. (2 weeks duration instead of 1 month)

heydemoura commented on client/jetpack-cloud/sections/partner-portal/unassign-license-dialog/style.scss:
```diff
@@ -0,0 +1,56 @@
+			font-weight: 600;
+			text-decoration: underline;
+		}
+	}
```
These styles are a little outdated. Our current production version has a different style for the dialog components on small screens.

Such as this revoke dialog:
![image](https://user-images.githubusercontent.com/5550190/182512085-b92c959e-ea82-41b3-b792-97e574242d48.png)

I would suggest rebasing this branch and applying the updated styles from the revoke dialog to this dialog as well.

Making those styles to be shared somehow would be nice, maybe having a higher Dialog component to apply these shared styles.

heydemoura replied:
We could skip the creation of a higher Dialog component for now and have it as a maintenance task to be looked at in the future, as we currently just have these 2 dialogs.

heydemoura reviewed (changes requested):
Code is looking good, just need some rebasing and updating some styles with updated versions from `trunk`.

taipeicoder committed:
Always enable passing vertical_id when the design type is vertical (#66198)

tyxla committed:
Framework: Refactor away from _.isPlainObject() (#65680)

ivan-ottinger committed:
Remove translation fallback from Import step (#66205)

flootr committed:
`PurchaseDetail`: refactor tests to `@testing-library/react` (#66118)

flootr committed:
`CountrySpecificPaymentFields`: refactor tests to `@testing-library/react` (#66166)

* `CountrySpecificPaymentFields`: refactor tests to `@testing-library/react`

* replace `.toHaveBeenNthCalledWith` with `.toHaveBeenCalledWith`

Co-authored-by: Marin Atanasov <8436925+tyxla@users.noreply.github.com>

* check each field for `disabled` status

Co-authored-by: Marin Atanasov <8436925+tyxla@users.noreply.github.com>

p-jackson committed:
Site status displayed in column rather than with badges (#66214)

p-jackson committed:
Sites Management Page: Tweak page styles (#66221)

danielbachhuber committed:
Sites Management Page: Use sentence case for 'New site' button (#66222)

heavyweight committed:
Happychat Connection: Update Tests (#65920)

* Happychat Connection: Update Tests

Move tests to appropriate package, minor refactoring, and add coverage for buildConnectionForCheckingAvailability

* Happychat Connection: Improve setup for tests

markbiek committed:
Prevent unnecessary renders in PurchaseNotice (#66138)

* Prevent unnecessary renders in PurchaseNotice

* Provide appropriate dependencies to `useEffect`
* Switch from `useState` to `useRef`

* Remove reference to Pro plan

WBerredo committed:
Update the sidebar height to fit-content. (#66093)

sejas committed:
Sites Dashboard Page: replace history when search or status change (#66174)

flootr committed:
`PlanDetails`: refactor tests to `@testing-library/react` (#66226)

mreishus committed:
theme: Use isThemePremium selector to determine if a theme is premium (#66170)

WBerredo committed:
Prefix the plugins cache keys with a version number (#66097)

mreishus committed:
theme: use isThemePurchased selector to detect purchase (#66173)

WBerredo committed:
Plugin Details: Manage plugin actions (#66054)

* Create ActivationButton component and use it as Plugin Details CTA

* Add plugin autoupdate toggle to the Plugin Details CTA area

* Create Manage Plugin Menu and add it to he plugin details CTA area

* Keep showing the placeholder until the request for sites in plugins is done.

* Update the autoupdates text

sirbrillig committed:
Checkout: Make Jetpack masterbar logo visible on pending page (#66141)

* Fallback to primary site in Layout if no selected site

* Show Jetpack masterbar colors on checkout pending section also

* Use selectedOrPrimarySiteId to determine isJetpack

* Change checkout masterbar css to use dedicated class for jetpack

* Remove changes to Layout

* Change other selectors to also use the new masterbar classnames

Copons committed:
Marketplace: Update Preinstalled Premium Plugins Browser Item (#65252)

- Fix preinstalled premium plugins labelled as inactive on simple sites.
- Show the installed count in the All Sites view.
- Fix the price label to clarify that the final cost might be higher.

mreishus committed:
theme: Replace price test with a new themeNeedsPurchase variable (#66183)

wilmersondasilva committed:
Merge branch 'add/unassign-license-feature' of github.com:Automattic/wp-calypso into add/unassign-license-feature

wilmersondasilva commented:
I closed this PR because a rebase operation went wrong, and it added a lot of commits that were not supposed to be in this PR. I'll create a new one containing only the correct commits.
